generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationConfigValidator.java

```diff
@@ -19,6 +23,18 @@ public ValidationResult validateCommandLine(GenerationConfig config) {
             errorMessages.add("RANDOM mode requires max row limit: use -n=<row limit> option");
         }
 
+        if (outputFileAlreadyExists(fileOutputTarget)) {
+            errorMessages.add("Output file already exists, please use a different output filename");
+        }
+
         return validationResult;
     }
+
+    private boolean outputFileAlreadyExists(OutputTarget fileOutputTarget) {
+        if (fileOutputTarget instanceof FileOutputTarget) {
```

Comment:
This feels bad, not sure what it could be 'replaced' with.
Is there only one implementation of an `OutputTarget` in production code?

Is there the remit to have an interface for 'file' output-targets, with an `exists()` (and `withFileName()`) methods?

Reply:
yes, it was bad, updated. OutputTargets have been refactored to allow testing to work.